bzt/resources/locustio-taurus-wrapper.py

```diff
@@ -5,70 +5,91 @@
 import time
 from requests.exceptions import HTTPError
 from locust import main, events, runners
+from locust.exception import StopLocust
 
 
-def getrec(request_type, name, response_time, response_length, exc=None):
-    rcode = '200' if exc is None else '500'
-    rmsg = 'OK' if exc is None else '%s' % exc
-    if isinstance(exc, HTTPError):
-        rcode = exc.message[:exc.message.index(' ')]
-        rmsg = exc.message[exc.message.index(':') + 2:]
+class LocustStarter(object):
+    def __init__(self):
+        super(LocustStarter, self).__init__()
+        self.fhd = None
+        self.writer = None
+        if os.getenv("LOCUST_DURATION"):
+            self.locust_start_time = time.time()
+            self.locust_duration = float(os.getenv("LOCUST_DURATION"))
+        else:
+            self.locust_start_time = None
+            self.locust_duration = None
 
-    return {
-        'timeStamp': "%d" % (time.time() * 1000),
-        'label': name,
-        'method': request_type,
-        'elapsed': response_time,
-        'bytes': response_length,  # NOTE: not sure if the field name is right
-        'responseCode': rcode,
-        'responseMessage': rmsg,
-        'success': 'true' if exc is None else 'false',
+    def __check_duration(self):
+        if self.locust_duration is not None:
+            if time.time() - self.locust_start_time >= self.locust_duration:
+                raise StopLocust
```

Comment:
Turns out this isn't nesessary. But i propose to use something like `StopLocust("Duration limit reached")` anyway, since it can be useful for users that read locust logs.